src/util/hass-translation.html

```diff
@@ -0,0 +1,100 @@
+<link rel='import' href='../../build-temp/translationFingerprints.html' />
+<link rel='import' href='../../build-temp/fallbackTranslation.html' />
+
+<script>
+function getActiveTranslation() {
+  // Perform case-insenstive comparison since browser isn't required to
+  // report languages with specific cases.
+  var lookup = {};
+  /* eslint-disable no-undef */
+  Object.keys(translationFingerprints).forEach(function (tr) {
+    lookup[tr.toLowerCase()] = tr;
+  });
+
+  // Search for a matching translation from most specific to general
+  function languageGetTranslation(language) {
+    var subtags = language.toLowerCase().split('-');
+
+    for (var i = subtags.length; i >= 1; i--) {
+      var lang = subtags.slice(0, i).join('-');
+
+      if (lookup[lang]) {
+        return lookup[lang];
+      }
+    }
+    return null;
+  }
+
+  var translation = null;
+  if (window.localStorage.language) {
+    translation = languageGetTranslation(window.localStorage.language);
+    if (translation) {
+      return translation;
+    }
+  } else if (navigator.languages) {
+    for (var i = 0; i < navigator.languages.length; i++) {
+      translation = languageGetTranslation(navigator.languages[i]);
+      if (translation) {
+        return translation;
+      }
+    }
+  } else {
+    translation = languageGetTranslation(navigator.language || navigator.userLanguage);
+    if (translation) {
+      return translation;
+    }
+  }
+
+  // Final fallback
+  return 'en';
+}
+
+// Store loaded translations in memory so translations are available immediately
+// when DOM is created in Polymer. Even a cache lookup creates noticable latency.
+var translations = {};
+
+window.getTranslation = function () {
+  var translation = getActiveTranslation();
+  /* eslint-disable no-undef */
+  var translationHash = translation + '-' + translationFingerprints[translation];
+
+  // If translation is available in memory, load it immediately
+  if (translations[translationHash]) {
```

Comment:
We should store the promise in `translations[translationHash]`. That way doing two requests at the same time won't fire off two requests.

Doing that means you will also have to remove the promise from the cache when it fails.